### PR TITLE
EZP-29999: As an editor, I want to see custom icons for content types

### PR DESCRIPTION
**Pull request by tischsoic**
JIRA: https://jira.ez.no/browse/EZP-29999

## Multi File Upload (MFU)
screen shot 2019-01-22 at 15 07 52

When it comes to MFU, we no longer display icon based on file type from _<input>_ but rather based on Contet Type returned from server. During loading (before receiving `Contnet.ContentType` from backend) we display no icon. The same goes for file which failed to be uploaded:

screen shot 2019-01-22 at 15 13 42

## UDW:
screen shot 2019-01-21 at 11 39 26

screen shot 2019-01-21 at 11 46 50

screen shot 2019-01-21 at 11 41 50

screen shot 2019-01-21 at 11 42 54

## Sub-items:

List view:
screen shot 2019-01-21 at 11 43 14

Grid view:
screen shot 2019-02-01 at 13 34 24

### CODE REVIEW

**Comment by sunpietro** [comment]
(Based on PR description) One thing to consider. The content type identifiers are set in the YAML config. So you don't need to wait for backend to get the content type identifier to display, because you already have one.

File: Resources/public/scss/modules/universal-discovery/_content.table.header.scss [hunk 1]
```diff
@@ -6,13 +6,18 @@
...
     &__list-header {
         padding: calculateRem(8px) calculateRem(16px);
-        font-weight: normal;
+        font-weight: 700;
         font-size: calculateRem(15px);
         line-height: calculateRem(20px);
         background-color: $ez-white;
 
+        &--icon {
+            flex: 0 0 58px;
```
Comment (sunpietro) [inline Resources/public/scss/modules/universal-discovery/_content.table.header.scss]: `calculateRem(58px)`

File: Resources/public/scss/modules/universal-discovery/_content.table.header.scss [hunk 2]
```diff
@@ -6,13 +6,18 @@
...
+        }
+
         &--name {
             flex: 1 1 calculateRem(280px);
+            padding-left: 10px;
```
Comment (sunpietro) [inline Resources/public/scss/modules/universal-discovery/_content.table.header.scss]: `calculateRem(10px)`

**Comment by sunpietro** [comment]
Actually, I was wrong. There was type detection inside the code.

File: src/modules/sub-items/components/table-view/table.view.component.js [hunk 3]
```diff
@@ -138,7 +138,8 @@ export default class TableViewComponent extends Component {
                     <td className={cellHeadClass}>
                         <input type="checkbox" checked={anyLocationSelected} onChange={this.selectAll} />
                     </td>
-                    <td className={`${cellHeadClass} ${cellClass}--name ${cellSortClass}`} onClick={this.sortByName} tabIndex="-1">
+                    <td className={cellHeadClass} />
+                    <td className={`${cellHeadClass} ${cellClass}--name ${cellSortClass}`} onClick={this.sortByName}  tabIndex="-1">
```
Comment (sunpietro) [inline src/modules/sub-items/components/table-view/table.view.component.js]: 2 many spaces after `className=""`

File: src/modules/universal-discovery/components/content-table/content.table.header.component.js [hunk 4]
```diff
@@ -6,6 +6,7 @@ const ContentTableHeaderComponent = () => {
 
     return (
         <div className="c-content-table-header__list-headers">
+            <div className="c-content-table-header__list-header c-content-table-header__list-header--icon" />
```
Comment (sunpietro) [inline src/modules/universal-discovery/components/content-table/content.table.header.component.js:9]: Empty div?

**Review by sunpietro — CHANGES_REQUESTED** [review]

File: src/modules/universal-discovery/components/content-table/content.table.header.component.js (hunk 4 above)
Comment (tischsoic) [reply src/modules/universal-discovery/components/content-table/content.table.header.component.js:9]: Yes, it's table-like structure, so we want to have header for all columns. Icons have separate column so they have also cell in header. However, we don't want to have any label in header for icons' column thus _div_ is empty.

**Comment by tischsoic** [comment]
Branch was rebased onto master and is ready for review.\
I applied fixes according to old review. ;) ping @sunpietro

File: Resources/public/scss/modules/sub-items-list/_table.view.item.scss [hunk 5]
```diff
@@ -33,6 +33,30 @@
...
+
+        &--icon {
+            padding: 0 calculateRem(8px) 0 calculateRem(16px);
+            width: calculateRem(24px);
+            text-align: right;
+            vertical-align: middle;
+            line-height: 0;
+        }
+
+        &--icon .ez-icon {
```
Comment (dew326) [inline Resources/public/scss/modules/sub-items-list/_table.view.item.scss]: Can be moved inside the previous `&--icon`.

File: Resources/public/scss/modules/sub-items-list/_table.view.item.scss [hunk 6]
```diff
@@ -33,6 +33,30 @@
...
+            fill: $ez-black;
+        }
+
+        &--name {
+            padding-left: 0;
+        }
+
+        &--priority {
+            width: 80px;
```
Comment (dew326) [inline Resources/public/scss/modules/sub-items-list/_table.view.item.scss]: calculateRem

File: Resources/public/scss/modules/sub-items-list/_table.view.item.scss [hunk 7]
```diff
@@ -33,6 +33,30 @@
...
+        }
+
+        &--actions {
+            width: 100px;
```
Comment (dew326) [inline Resources/public/scss/modules/sub-items-list/_table.view.item.scss]: calculateRem

File: src/modules/multi-file-upload/components/upload-list/upload.item.component.js [hunk 8]
```diff
@@ -337,24 +337,43 @@ export default class UploadItemComponent extends Component {
...
-            return 'image';
+    getContentTypeIdentifier() {
+        const { contentTypesMap, data } = this.props;
+
+        if (!data.struct || !data.struct.Content) {
+            return null
         }
 
-        return 'file';
+        const contentTypeHref = data.struct ? data.struct.Content.ContentType._href : null;
```
Comment (dew326) [inline src/modules/multi-file-upload/components/upload-list/upload.item.component.js]: I believe `data.struct` is available as you have early return.

File: src/modules/multi-file-upload/components/upload-list/upload.item.component.js [hunk 9]
```diff
@@ -337,24 +337,43 @@ export default class UploadItemComponent extends Component {
...
+        const contentTypeIdentifier = contentType ? contentType.identifier : null;
+
+        return contentTypeIdentifier;
+    }
+
+    /**
+     * Renders an icon of a content type
+     *
+     * @method renderIcon
+     * @returns {JSX.Element}
```
Comment (dew326) [inline src/modules/multi-file-upload/components/upload-list/upload.item.component.js]: not always true?

File: src/modules/multi-file-upload/components/upload-list/upload.item.component.js [hunk 10]
```diff
@@ -337,24 +337,43 @@ export default class UploadItemComponent extends Component {
...
 
     /**
-     * Detects a file type of uploaded file
+     * Returns content type identifier 
+     * based on Content object returned from server after upload
      *
-     * @method detectFileType
+     * @method getContentTypeIdentifier
      * @memberof UploadItemComponent
      * @returns {String}
```
Comment (dew326) [inline src/modules/multi-file-upload/components/upload-list/upload.item.component.js]: not always true?

File: src/modules/universal-discovery/components/content-meta-preview/content.meta.preview.component.js [hunk 11]
```diff
@@ -15,7 +15,7 @@ export default class ContentMetaPreviewComponent extends Component {
      * Renders an icon related to a content type
      *
      * @method renderIcon
-     * @returns {Element}
+     * @returns {JSX.Element}
```
Comment (dew326) [inline src/modules/universal-discovery/components/content-meta-preview/content.meta.preview.component.js]: not always true?

File: src/modules/universal-discovery/components/finder/finder.tree.leaf.component.js [hunk 12]
```diff
@@ -27,19 +27,55 @@ export default class FinderTreeLeafComponent extends Component {
...
+        const contentTypeIdentifier = contentType ? contentType.identifier : null;
+
+        return contentTypeIdentifier;
+    }
+
+    /**
+     * Renders an icon of a content type
+     *
+     * @method renderIcon
+     * @returns {JSX.Element}
```
Comment (dew326) [inline src/modules/universal-discovery/components/finder/finder.tree.leaf.component.js]: not always true?

File: src/modules/universal-discovery/components/finder/finder.tree.leaf.component.js [hunk 13]
```diff
@@ -27,19 +27,55 @@ export default class FinderTreeLeafComponent extends Component {
...
+            </div>
+        );
+    }
+
     /**
      * Renders a loading state icon
      *
      * @method renderLoadingIcon
-     * @returns {Element}
+     * @returns {JSX.Element}
```
Comment (dew326) [inline src/modules/universal-discovery/components/finder/finder.tree.leaf.component.js]: not always true?

**Review by dew326 — CHANGES_REQUESTED** [review]

**Comment by tischsoic** [comment]
Ping @dew326 @sunpietro

**Review by dew326 — APPROVED** [review]

**Review by sunpietro — APPROVED** [review]

**Review by barbaragr — APPROVED** [review]

**Pull request merged**